Subject: disable for ppc32: [SCSI] spi transport: don't allow dt to be set on SE or HVD buses
From: olh@suse.de
Patch-mainline: 2.6.19

This patch causes machine checks on pmac:

http://www.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=60eef25701d25e99c991dd0f4a9f3832a0c3ad3e

Add a delay before accessing registers.

Signed-off-by: Olaf Hering <olh@suse.de>
---
 drivers/scsi/aic7xxx/aic7xxx_osm.c |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

Index: linux-2.6.18-rc7/drivers/scsi/aic7xxx/aic7xxx_osm.c
===================================================================
--- linux-2.6.18-rc7.orig/drivers/scsi/aic7xxx/aic7xxx_osm.c
+++ linux-2.6.18-rc7/drivers/scsi/aic7xxx/aic7xxx_osm.c
@@ -2539,7 +2539,14 @@ static void ahc_linux_set_iu(struct scsi
 static void ahc_linux_get_signalling(struct Scsi_Host *shost)
 {
 	struct ahc_softc *ahc = *(struct ahc_softc **)shost->hostdata;
-	u8 mode = ahc_inb(ahc, SBLKCTL);
+	unsigned long flags;
+	u8 mode;
+
+	ahc_lock(ahc, &flags);
+	ahc_pause(ahc);
+	mode = ahc_inb(ahc, SBLKCTL);
+	ahc_unpause(ahc);
+	ahc_unlock(ahc, &flags);
 
 	if (mode & ENAB40)
 		spi_signalling(shost) = SPI_SIGNAL_LVD;
